Classes/DDFileLogger.m

```diff
@@ -905,128 +933,152 @@ - (void)maybeRollLogFileDueToSize {
+            self->_currentLogFileInfo.isArchived = YES;
+            NSString *archivedLogFilePath = [self->_currentLogFileInfo.fileName copy];
+            self->_currentLogFileInfo = nil;
 
-                self->_currentLogFileInfo = [[DDLogFileInfo alloc] initWithFilePath:currentLogFilePath];
+            if ([self->_logFileManager respondsToSelector:@selector(didArchiveLogFile:)]) {
+                dispatch_async(_completionQueue, ^{
+                    [self->_logFileManager didArchiveLogFile:archivedLogFilePath];
+                });
             }
         }
+    }
 
-        result = self->_currentLogFileInfo;
-    });
+    if (isResuming && self->_currentLogFileInfo) {
+        NSLogVerbose(@"DDFileLogger: Resuming logging with file %@", self->_currentLogFileInfo.fileName);
+    }
 
-    return result;
+    if (!self->_currentLogFileInfo) {
+        NSString *currentLogFilePath = [self->_logFileManager createNewLogFile];
+        self->_currentLogFileInfo = [[DDLogFileInfo alloc] initWithFilePath:currentLogFilePath];
+    }
+
+    return self->_currentLogFileInfo;
 }
 
-- (NSFileHandle *)currentLogFileHandle {
-    __block NSFileHandle *localCurrentLogFileHandle;
-    
-    dispatch_sync(self.currentLogFileHandleQueue, ^{
-        if (self->_currentLogFileHandle == nil) {
-            NSString *logFilePath = [[self currentLogFileInfo] filePath];
-            
-            self->_currentLogFileHandle = [NSFileHandle fileHandleForWritingAtPath:logFilePath];
-            [self->_currentLogFileHandle seekToEndOfFile];
-            
-            if (self->_currentLogFileHandle) {
-                [self scheduleTimerToRollLogFileDueToAge];
-                
-                // Here we are monitoring the log file. In case if it would be deleted ormoved
-                // somewhere we want to roll it and use a new one.
-                self->_currentLogFileVnode = dispatch_source_create(
-                                                              DISPATCH_SOURCE_TYPE_VNODE,
-                                                                    (uintptr_t)[self->_currentLogFileHandle fileDescriptor],
-                                                              DISPATCH_VNODE_DELETE | DISPATCH_VNODE_RENAME,
-                                                              self.loggerQueue
-                                                              );
-
-                __weak __auto_type weakSelf = self;
-                dispatch_source_set_event_handler(self->_currentLogFileVnode, ^{ @autoreleasepool {
-                    NSLogInfo(@"DDFileLogger: Current logfile was moved. Rolling it and creating a new one");
-                    [weakSelf rollLogFileNow];
-                } });
-                
+- (void)lt_monitorCurrentLogFileForExternalChanges {
+    NSAssert(self->_currentLogFileHandle, @"Can not monitor without handle.");
+    NSAssert([self isOnInternalLoggerQueue], @"lt_ methods should be on logger queue.");
```

Comment:
Nit-Pick: Logically we should first assert being on the internal queue and then access the currentLogFileHandle.

Reply:
good catch